wallet/src/ui/app/pages/dapp-tx-approval/index.tsx

```diff
@@ -67,56 +184,80 @@ export function DappTxApprovalPage() {
                       { label: 'Contents', content: txRequest?.txBytes },
                   ],
         [txRequest]
     );
+
     return (
         <Loading loading={loading}>
             {txRequest ? (
                 <UserApproveContainer
-                    title="Transaction Request"
                     origin={txRequest.origin}
                     originFavIcon={txRequest.originFavIcon}
                     approveTitle="Approve"
                     rejectTitle="Reject"
                     onSubmit={handleOnSubmit}
                 >
-                    {valuesContent.map(({ label, content }) => (
-                        <Fragment key={label}>
-                            <label className={stUserApprove.label}>
-                                {label}
-                            </label>
-                            <div className={st.value}>{content}</div>
-                        </Fragment>
-                    ))}
+                    <dl className={st.card}>
+                        <div className={st.content}>
+                            {valuesContent.map(({ label, content }) => (
+                                <div key={label} className={st.row}>
+                                    <dt>{label}</dt>
+                                    <dd>{content}</dd>
+                                </div>
+                            ))}
+                        </div>
+                    </dl>
+                    {metadata && tab && (
+                        <>
+                            <div className={st.tabs}>
+                                {Object.entries(metadata).map(
+                                    ([key, value]) =>
+                                        value.children.length > 0 && (
+                                            <button
+                                                type="button"
+                                                className={cl(
+                                                    st.tab,
+                                                    tab === key && st.active
+                                                )}
+                                                // eslint-disable-next-line react/jsx-no-bind
+                                                onClick={() => {
+                                                    setTab(key as TabType);
+                                                }}
+                                            >
+                                                {value.name}
+                                            </button>
+                                        )
+                                )}
+                            </div>
+                            <div className={st.objects}>
+                                {metadata[tab].children.map(
+                                    ({ id, module }, index) => (
+                                        <PassedObject
+                                            key={index}
```

Comment:
Why not using the id as a key?

Reply:
I need to figure out if it is required to be unique. E.g. can I pass the same object to a function twice.

Reply:
I need to figure out if it is required to be unique. E.g. can I pass the same object to a function twice.

Reply:
Good point, I don't think there is something forbidding it (but that's my assumption)